### PR TITLE
[PHP 8.1] fix Passing null to parameter #1 ($string) of type string is deprecate

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
https://github.com/coreshop/CoreShop/runs/6981738951?check_suite_focus=true#step:17:93

### CODE REVIEW

**Comment by blankse** [comment]
The $siteId attribute should as default value have a empty array. So it cant be null in the future (Maybe the support can be dropped in Pimcore 11).

**Comment by dvesh3** [comment]
> The $siteId attribute should as default value have a empty array. So it cant be null in the future (Maybe the support can be dropped in Pimcore 11).

@dpfaffenbauer could you please update it? thanks!

**Comment by dpfaffenbauer** [comment]
@dvesh3 @blankse changed. But I don't know why it is null... Maybe cause how it is stored in Config Store or something like this... So IMO that will not really solve the problem, since it still can be null for now.

**Comment by blankse** [comment]
@dpfaffenbauer I mean the attribute in line 69, not the param. So if you create a new static route and save it it can't be null and the load method set null to the setSiteId method.

**Comment by dpfaffenbauer** [comment]
@blankse gotcha

File: models/Staticroute.php [hunk 1]
```diff
@@ -394,13 +394,17 @@ public function getReverse()
...
+     * @param string|array|null $siteId
      *
      * @return $this
      */
     public function setSiteId($siteId)
     {
         $result = [];
+        
+        if (null === $siteId) {
+            return;
```
Comment (blankse) [inline models/Staticroute.php]: I think this could stay. Null should set an empty array.
```suggestion
            $siteId = [];
```

**Comment by dvesh3** [comment]
@dpfaffenbauer @blankse thanks guys!!

**Pull request merged**